lldb/source/Interpreter/CommandInterpreter.cpp

```diff
@@ -518,6 +518,15 @@ void CommandInterpreter::Initialize() {
     AddAlias("re", cmd_obj_sp);
   }
 
+  cmd_obj_sp = GetCommandSPExact("scripting execute");
+  if (cmd_obj_sp) {
+    AddAlias("sc", cmd_obj_sp);
+    AddAlias("scr", cmd_obj_sp);
+    AddAlias("scri", cmd_obj_sp);
+    AddAlias("scrip", cmd_obj_sp);
+    AddAlias("script", cmd_obj_sp);
```

Comment:
this is unfortunate, do these pollute help output? if so, can we hide them?

is there no way to make a prefix match an alias, when it's shorter?

Reply:
The problem is that if we have the command `scripting` and the alias `script` and we see `sc` we need to resolve that command ambiguity.  The only tool we have for ambiguous command resolution at present is "exact matches always win".

If we wanted to solve this w/o making all the exact matches we intend to win, we could add a ranking system to the commands and aliases, which bias ambiguous matches in favor of one or the other of the commands, or some similar system.  I worry that would get messy pretty quickly, but anyway, you'd need something like that.

Reply:
I agree its not the most elegant solution but I'd prefer solving the partial matching issue in a follow-up.

Reply:
+1